app/src/main/java/com/yuneec/example/component/actitivty/MainActivity.java

```diff
@@ -184,16 +181,25 @@ public void run() {
                 if (!result.equals("Success")) {
                     Common.makeToast(context, "Please make sure SD card is inserted and try again");
                 } else {
-                    if (mode.equals(Camera.Mode.PHOTO)) {
+                    if (mode.equals(Camera.Mode.PHOTO) && !CameraSettingsFragment.isPhotoInterval) {
                         Camera.asyncTakePhoto();
-                    } else {
-                        video = (Button) findViewById(R.id.video);
-                        if (video.getText().equals("Start Video")) {
+                    } else if (mode.equals(Camera.Mode.VIDEO)) {
+                        mediaCapture = (Button) findViewById(R.id.video);
+                        if (mediaCapture.getText().equals(getString(R.string.video))) {
                             Camera.asyncStartVideo();
-                            video.setText("Stop Video");
+                            mediaCapture.setText(getString(R.string.stop_video));
                         } else {
                             Camera.asyncStopVideo();
-                            video.setText("Start Video");
+                            mediaCapture.setText(getString(R.string.video));
+                        }
+                    } else {
+                        mediaCapture = (Button) findViewById(R.id.photo_interval);
+                        if (mediaCapture.getText().equals(getString(R.string.photo_interval))) {
```

Comment:
Potentially, we should listen the camera status and decide based on that what button text to show. Otherwise, we might get out of sync if some error occurs on the camera side.

I think the way to do that would be to register callbacks to get the status and then trigger a UI update when a status comes in.

Reply:
Done.

Reply:
Oh nice.